flatdata-py/flatdata/lib/archive_builder.py

```diff
@@ -0,0 +1,290 @@
+'''
+ Copyright (c) 2021 HERE Europe B.V.
+ See the LICENSE file in the root of this project for license details.
+'''
+
+from collections import namedtuple
+import os
+
+from .errors import *
+from .resources import *
+from .data_access import write_value
+
+_SCHEMA_EXT = ".schema"
+
+ResourceSignature = namedtuple("ResourceSignature",
+                               ["container", "initializer", "schema", "is_optional", "doc"])
+
+
+class IndexWriter:
+    """
+    IndexWriter class. Only applicable when multivector is present in archive schema.
+    """
+
+    def __init__(self, name, size, resource_storage):
+        """
+        Create IndexWriter class.
+
+        All arguments are required.
+        """
+        if not (name and resource_storage and size):
+            raise IndexWriterError(
+                f"Either ResourceStorage: {resource_storage} or name: {name} or size:"
+                 "{size} not provided.")
+
+        self._name = name
+        self._index_size = size
+        self._fout = resource_storage.get(f'{self._name}_index', False)
+
+    def add(self, index):
+        """
+        Convert index(number) to bytearray and add to in memory store
+        """
+        index_bytes = int(index).to_bytes(self._index_size,
+                                          byteorder="little", signed=False)
+        self._fout.write(index_bytes)
+
+    def finish(self):
+        """
+        Complete index resource by adding size and padding followed by writing to file
+        """
+        self._fout.add_size()
+        self._fout.add_padding()
+        self._fout.close()
```

Comment:
PS: Is there anything that will throw an error in case any of the writing steps fail?

Reply:
Underlying classes will handle exceptions